packages/edit-navigation/src/components/menu-editor/index.js

```diff
@@ -6,66 +6,81 @@ import {
 	BlockEditorProvider,
 } from '@wordpress/block-editor';
 import { useViewportMatch } from '@wordpress/compose';
-import { useMemo } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import useMenuItems from './use-menu-items';
-import useNavigationBlocks from './use-navigation-blocks';
 import MenuEditorShortcuts from './shortcuts';
 import BlockEditorArea from './block-editor-area';
 import NavigationStructureArea from './navigation-structure-area';
+import useNavigationBlockEditor from './use-navigation-block-editor';
+import useStubPost from './use-stub-post';
+import { useMemo } from '@wordpress/element';
 
 export default function MenuEditor( {
 	menuId,
 	blockEditorSettings,
 	onDeleteMenu,
 } ) {
-	const isLargeViewport = useViewportMatch( 'medium' );
 	const query = useMemo( () => ( { menus: menuId, per_page: -1 } ), [
 		menuId,
 	] );
-	const {
-		menuItems,
-		eventuallySaveMenuItems,
-		createMissingMenuItems,
-	} = useMenuItems( query );
-	const { blocks, setBlocks, menuItemsRef } = useNavigationBlocks(
-		menuItems
-	);
-	const saveMenuItems = () => eventuallySaveMenuItems( blocks, menuItemsRef );
+	const postId = useStubPost( query );
```

Comment:
Instead of keeping blocks in state, this PR experiments with creating a "stub" post that serves as container for all our blocks but is never meant to be saved.